assets/queries/terraform/kubernetes_pod/service_account_name_undefined_or_empty/query.rego

```diff
@@ -0,0 +1,39 @@
+package Cx
+
+CxPolicy[result] {
+	resource := input.document[i].resource.kubernetes_pod[name]
+
+	spec := resource.spec
+    object.get(spec, "service_account_name", "undefined") == "undefined"
+
+	result := {
+		"documentId": input.document[i].id,
+		"searchKey": sprintf("kubernetes_pod[%s].spec", [name]),
+		"issueType": "MissingAttribute",
+		"keyExpectedValue": sprintf("kubernetes_pod[%s].spec.service_account_name is defined", [name]),
+		"keyActualValue": sprintf("kubernetes_pod[%s].spec.service_account_name is undefined", [name]),
+	}
+}
+
+CxPolicy[result] {
+	resource := input.document[i].resource.kubernetes_pod[name]
+
+	service_account_name := resource.spec.service_account_name
+    checkIsNullOrEmpty(service_account_name)
```

Comment:
Why don't you replace this function with a direct assertion?
`service_account_name == ["", null][j]`